lib/action_view/component/base.rb

```diff
@@ -110,40 +110,41 @@ def call_method_name(variant)
           end
         end
 
-        def has_initializer?
-          self.instance_method(:initialize).owner == self
-        end
-
         def source_location
-          # Require #initialize to be defined so that we can use
-          # method#source_location to look up the file name
-          # of the component.
-          #
-          # If we were able to only support Ruby 2.7+,
-          # We could just use Module#const_source_location,
-          # rendering this unnecessary.
-          raise NotImplementedError.new("#{self} must implement #initialize.") unless has_initializer?
-
-          instance_method(:initialize).source_location[0]
+          @source_location ||=
```

Comment:
Caching this reflection seems like a great idea. We call this a bunch of times when compiling a template.